tests/test_utils.py

```diff
@@ -168,3 +171,66 @@ def test_make_clip_df_from_label_df(silence_10s_mp3_str, metadata_wav_str):
     # should copy labels for each file to all clips of that file
     # duplicate file should have labels from _first_ occurrence in label_df
     assert np.array_equal(clip_df["a"].values, [0, 0, 0, 0, 2, 2])
+    
+    
+
+# The @pytest.mark.parametrize decorator loops trough each value in list when running pytest. 
+# If you add --verbose, it also prints if it passed for each value in the list for each function 
+# that takes it as input.
+
+@pytest.mark.parametrize("input", [1, 11, 13, 42, 59, 666, 1234]) 
+def test_torch_rand(input):
+    utils.set_seed(input)
+    tr1 = torch.rand(100)
+
+    utils.set_seed(input)
+    tr2 = torch.rand(100)
+
+    utils.set_seed(input + 1)
+    tr3 = torch.rand(100)
+
+    assert all(tr1 == tr2) & any(tr1 != tr3)
+
+@pytest.mark.parametrize("input", [1, 11, 13, 42, 59, 666, 1234])
+def test_numpy_random_rand(input):
+    utils.set_seed(input)
+    nr1 = np.random.rand(100)
+
+    utils.set_seed(input)
+    nr2 = np.random.rand(100)
+
+    utils.set_seed(input + 1)
+    nr3 = np.random.rand(100)
+
+    assert all(nr1 == nr2) & any(nr1 != nr3)
+
+@pytest.mark.parametrize("input", [1, 11, 13, 42, 59, 666, 1234])
+def test_radom_sample(input):
+    list1000 = list(range(1, 1000))
+
+    utils.set_seed(input)
+    rs1 = random.sample(list1000, 100)
+
+    utils.set_seed(input)
+    rs2 = random.sample(list1000, 100)
+
+    utils.set_seed(input + 1)
+    rs3 = random.sample(list1000, 100)
+
+    assert (rs1 == rs2) & (rs1 != rs3)
+
+@pytest.mark.parametrize("input", [1, 11, 13, 42, 59, 666, 1234])
+def test_cnn(input):
+    utils.set_seed(input)
+    model_resnet1 = cnn_architectures.resnet18(num_classes=10, weights=None)
+    lw1 = model_resnet1.layer1[0].conv1.weight
+
+    utils.set_seed(input)
+    model_resnet2 = cnn_architectures.resnet18(num_classes=10, weights=None)
+    lw2 = model_resnet2.layer1[0].conv1.weight
+
+    utils.set_seed(input + 1)
+    model_resnet3 = cnn_architectures.resnet18(num_classes=10, weights=None)
+    lw3 = model_resnet3.layer1[0].conv1.weight
+
+    assert torch.all(lw1 == lw2) & torch.any(lw1 != lw3)
```

Comment:
don't understand the logic here, can you add a comment? is the `any` asserting that _none_ of the weights should be the same? That doesn't seem like the correct logic, rather we want to assert that the entire array isn't equal to the other?

Reply:
In summary, also assert that at least one element is different if different seeds. Added as a comment in [tests/test_utils.py](https://github.com/kitzeslab/opensoundscape/blob/1fc6d3a00197f9141ec5d92efdbf4d5bb4fe81bd/tests/test_utils.py#L181).